### ISSUE TITLE
Using Github discussions instead of mailing list and IRC

### ISSUE THREAD

**Issue by PolarGoose** [skvadrik/re2c#512]
[Github discussions](https://docs.github.com/en/discussions) can be a better alternative than a mailing list or IRC.
Has using this feature been considered? The discussions are Google searchable and more user-friendly.

**Comment by skvadrik** [comment]
Sure, why not. Here's a welcome discussion: https://github.com/skvadrik/re2c/discussions/513

**Comment by PolarGoose** [comment]
Amazing. Thank you.

By the way, why do you want to keep the mailing list and IRC?

**Comment by skvadrik** [comment]
> By the way, why do you want to keep the mailing list and IRC?

IRC channels are occasionally used. Mailing lists haven't been used for a long time (people usually email me directly), but they don't require any maintenance.

**Comment by PolarGoose** [comment]
> but they don't require any maintenance.

Okay, that's reasonable. In my opinion, it is better to have one channel of communication that is visible to anyone and Google searchable.

But it is your choice. Maybe you will consider to deprecate the mailing list and IRC in the future.

You can close this ticket.

**Comment by skvadrik** [comment]
> In my opinion, it is better to have one channel of communication that is visible to anyone and Google searchable.

I agree with you that Discussions are better for the reasons you mentioned. But I think it's best to keep all options. Some people feel uncomfortable communicating in English, or they don't want to communicate in public, or they have got used to IRC over the years, or for some other reason they prefer IRC / email.

> Maybe you will consider to deprecate the mailing list and IRC in the future.

We should definitely add Discussions to the website as a preferred way of informal communications (I'll keep the bug open so I don't forget to do so).

**Comment by skvadrik** [comment]
> We should definitely add Discussions to the website as a preferred way of informal communications (I'll keep the bug open so I don't forget to do so).

I've mentioned them in the website sources and will update the website soon. Closing in preparation to release 4.2.
